prime-router/src/test/kotlin/azure/FHIRFlowFunctionsTests.kt

```diff
@@ -0,0 +1,127 @@
+                "{\"class\":\"gov.cdc.prime.router.engine.RawSubmission\"," +
+                    "\"blobURL\":\"http://localhost:10000/devstoreaccount1/reports/receive%2Fsimple_report.hl7test%2F" +
+                    "test-covid-19-4990af55-c25b-474b-b4f6-7ccf155706b0-20220210113202.hl7\"," +
+                    "\"digest\":\"badffffff9bffffffcb46fffffff5ffffff886fffffff84ffffff9efffffffaffffffd23bfffffff3" +
+                    "ffffff8d4d7ffffffff2156fffffffbaffffff963ffffffff6ffffffa0397b10607fffffffa36cffffffbf71\"," +
+                    "\"sender\":\"simple_report.hl7test\"}",
+                IllegalStateException(
+                    "FHIR - Downloaded file does not match expected file " +
+                        "badffffff9bffffffcb46fffffff5ffffff886fffffff84ffffff9efffffffaffffffd23bfffffff3ffffff8d4d7" +
+                        "ffffffff2156fffffffbaffffff963ffffffff6ffffffa0397b10607fffffffa36cffffffbf71" +
+                        " | " +
+                        "2bffffff90ffffffa22674fffffff713ffffffbd64ffffffb26c663426877fffffff6ffffffa91aeffffffd4" +
+                        "ffffff82313b5fffffffc1ffffffaf653a387323"
+                ),
+            ),
+        )
+
+        mockkObject(BlobAccess.Companion)
+        every { BlobAccess.Companion.downloadBlob(any()) } returns testHL7NonBulk.toByteArray()
+        val fhirEngine = FHIREngine()
+        testCases.forEach { test ->
+            try {
+                fhirEngine.process(test.message)
+                assertNull(test.exception, test.name)
+            } catch (t: Throwable) {
+                if (test.exception == null) {
+                    throw t
+                }
+                assertEquals(test.exception::class, t::class, test.name)
+                assertEquals(test.exception.message, t.message, test.name)
+            }
+        }
+    }
+
+    @Test
+    fun `test results`() {
+        data class TestCase(
+            val name: String,
+            val input: String,
+            val output: String,
+            val exception: Throwable? = null
+        )
+
+        val testCases = listOf(
+            TestCase(
+                "matching",
+                testHL7NonBulk,
+                testHL7NonBulk,
+            ),
+            TestCase(
+                "not matching",
+                testHL7NonBulk,
+                testOtherHL7NonBulk,
+                IllegalStateException("FHIR - HL7 processing failed")
+            )
+        )
+
+        val fhirEngine = FHIREngine()
+        testCases.forEach { case ->
+            try {
+                fhirEngine.compare(case.input, case.output)
+                assertNull(case.exception, case.name)
+            } catch (t: Throwable) {
+                if (case.exception == null) {
+                    throw Error(case.name, t)
+                }
+                assertEquals(case.exception::class, t::class, case.name)
+                assertEquals(case.exception.message, t.message, case.name)
```

Comment:
Change these to assertK if possible. You can also add extension methods to assertK if these don't transfer cleanly. If you can't transfer them, please add a comment so future maintainers know why these aren't using assertK.